### DISCUSSION TITLE
Tailwind to scan strings in JS files for build.css

### DISCUSSION THREAD

**Discussion by mrpbennett** [tailwindlabs/tailwindcss#7342] [Ideas]
I have a project that is using the old v2 CDN link. I have gone ahead and updated the project to use v3 however I have a JavaScript file that is using strings with Tailwind Styling, the build process doesn't seem to pick these up. For example I have this:

```js
let pixelInfo = `
<div class="element">
<div class="question flex justify-between px-6 py-4 ${colorGrade} ${borderColor} hvr-underline-from-left">
  <span class="text-sm ${textColor} whitespace-nowrap">${pixelType}</span>
  <button class="w-full flex flex-row-reverse">
	<i class="fas fa-plus-circle"></i>
  </button>
</div>

<div class="answer hideText">
  <span id="pixel-url" class="c-word-wrap text-sm ">${pixelUrl}</span>
	<span id="query-params">${queryParams}</span>
</div>
</div>
`;
```

This is my `tailwind.config.js`

```js
module.exports = {
  content: [
    "/background.js",
    "./popup/**/*.{html,js}",
    ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

Although the path is correct for `background.js` because they're strings with the style it doesn't seem to pick them up. I have tested this building my CSS file and then removing the CDN link. Where the styles fail 

Is there some sort of string parsing we could do with JS files?

**Comment by chexburger** [comment]
Having the same issue. I have a JavaScript file that is applying classes to created elements dynamically and those aren't picked up. Documentation says it scans the files for tailwind class names, but that is obviously not the case.

https://tailwindcss.com/docs/content-configuration#configuring-source-paths

**Reply by mrpbennett** [reply]
It works as intended I believe, as it looks for classes. But I think it doesn’t parse strings for classes,

**Reply by chexburger** [reply]
It would have to. I don't see how it would find any other classes. I have each class separated in quotes as required by the JavaScript I am using. It wouldn't need to split the string apart.

classList.add("Class") isn't detected, an array of classes isn't detected either.

It might only read the file, not parse it. Don't know why it wouldn't be able to find it then if it just read the file as text. Either way doesn't seem to detect the classes out of JavaScript.

**Reply by mrpbennett** [reply]
I would say your issue is different to the one I have. I’m after praising of template strings or strings in JS that hold CSS classes. 

Without more info or code examples would be hard to differentiate the two issues

**Comment by robbyrice** [comment]
As mentioned here [https://tailwindcss.com/docs/content-configuration#dynamic-class-names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) and here [https://tailwindcss.com/docs/content-configuration#class-detection-in-depth](https://tailwindcss.com/docs/content-configuration#class-detection-in-depth) Tailwind doesn't run your code and so you can't use dynamic class names or variables.

**Comment by boggye** [comment]
I have exactly the same issue as the original poster. It doesn't seem to pull the class names from the following code:
```
    positionTitleRenderer: function(value, metaData, record, rowIndex, colIndex, store, view) {
        return `<div class="text-ellipsis overflow-hidden text-slate-800"><span class="font-bold text-sm">${record.data.positionTitle}</span> - ${record.data.postingNumber}</div>`;
    }
```
Note that none of the TW classes are stored in variables. Any suggestions? 

Unless I missed it, it would have been very nice and useful if the utility had a debug command option to display the name of each that is scanned and the classes that it extracted from that file.

**Reply by wongjn** [reply]
Seems to work fine in [Tailwind Play](https://play.tailwindcss.com/AZSB24WXrF) (open the *Generated CSS* panel) which would point to a misconfiguration in the project. Consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by boggye** [reply]
@wongjn you just pasted the JavaScript code in the html section. Of course it works :-)

**Reply by wongjn** [reply]
That is how Tailwind works – it does not perceive any files in any special way. Rather, it scans all files covered by the `content` file globs as strings, whether those files contain HTML, JavaScript or any other language. Consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

You could also consider checking:
- Tailwind has compiled since you added the classes.
- The `content` file globs sufficiently cover the file that the code appears in.
- The class names would resolve to a Tailwind class that Tailwind would generate a rule for.

**Comment by boggye** [comment]
@mrpbennett Did you find a solution to this?

**Reply by mrpbennett** [reply]
I didn't I'm afraid.

**Comment by boggye** [comment]
ok, after more debugging, it turns out it was my fault, the path pointing to the js files was incorrect. It works - I also installed the latest version 3.3.3. 
@wongjn You were right, I tested my js file in the tw play and it worked.
